enclave/core/sgx/thread.c

```diff
@@ -242,8 +325,18 @@ oe_result_t oe_mutex_lock(oe_mutex_t* mutex)
         oe_spin_lock(&m->lock);
         {
             /* Attempt to acquire lock */
-            if (_mutex_lock(m, self) == 0)
+            oe_result_t result = _mutex_lock(m, self);
+
+            if (result == OE_BUSY)
+            {
+                /* Deadlock */
+                OE_TRACE_ERROR("Recursively acquire a non-recursive lock");
+                continue;
+            }
+
+            if (result == OE_OK)
             {
+                /* Successfully aquire the lock */
```

Comment:
This minor comment is for line 348 below, which is not included in this PR, but could you fix the indentation?

Reply:
Thanks for the review. Can you point me what's the exact line of code you refers to given that line number is changing over time.